custom_components/tahoma/climate_st.py

```diff
@@ -0,0 +1,242 @@
+        await super().async_added_to_hass()
+        async_track_state_change(
+            self.hass, self._temp_sensor_entity_id, self._async_temp_sensor_changed
+        )
+
+        @callback
+        def _async_startup(event):
+            """Init on startup."""
+            if self._temp_sensor_entity_id is not None:
+                temp_sensor_state = self.hass.states.get(self._temp_sensor_entity_id)
+                if temp_sensor_state and temp_sensor_state.state != STATE_UNKNOWN:
+                    self.update_temp(temp_sensor_state)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+
+        self.schedule_update_ha_state(True)
+
+    async def _async_temp_sensor_changed(self, entity_id, old_state, new_state) -> None:
+        """Handle temperature changes."""
+        if new_state is None or old_state == new_state:
+            return
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return MAP_HVAC_MODES[self.select_state(ST_DEROGATION_TYPE_STATE)]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return MAP_PRESET_MODES[
+            self.select_state(ST_HEATING_MODE_STATE)
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
+        ]
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return [
+            PRESET_NONE,
+            PRESET_FREEZE,
+            PRESET_NIGHT,
+            PRESET_AWAY,
+            PRESET_HOME,
+        ]
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 15.0
```

Comment:
Why?

Reply:
Because Somfy Smart Thermostat is limited to the range 15-26°C. As this property is part of the ClimateEntity, I think it's better to set that her, than check whether temperature is in the right range in `set_temperature` (although that should be done as well in case a user passes a temperature outside of the range in a service or an automation...)

Reply:
Got it. HA probably has a check on this property if we set a value below this value.